Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/CopyPasteHelper.cs

```diff
@@ -0,0 +1,442 @@
+            PasteAction(arrayProperty, buffer.ToString(), toDuplicate.GetIndexOfArrayElement() + 1);
+        }
+
+        public static int DuplicateBinding(SerializedProperty arrayProperty, SerializedProperty toDuplicate, string newActionName, int index)
+        {
+            if (IsComposite(toDuplicate))
+                return DuplicateComposite(arrayProperty, toDuplicate, PropertyName(toDuplicate), newActionName, index, out _).GetIndexOfArrayElement();
+            var binding = DuplicateElement(arrayProperty, toDuplicate, newActionName, index, false);
+            binding.FindPropertyRelative("m_Action").stringValue = newActionName;
+            return index;
+        }
+
+        private static SerializedProperty DuplicateComposite(SerializedProperty bindingsArray, SerializedProperty compositeToDuplicate, string name, string actionName, int index, out int newIndex, bool increaseIndex = true)
+        {
+            if (increaseIndex)
+                index += InputActionSerializationHelpers.GetCompositePartCount(bindingsArray, compositeToDuplicate.GetIndexOfArrayElement());
+            var newComposite = DuplicateElement(bindingsArray, compositeToDuplicate, name, index++, false);
+            newComposite.FindPropertyRelative("m_Action").stringValue = actionName;
+            var bindings = GetBindingsForComposite(bindingsArray, compositeToDuplicate.GetIndexOfArrayElement());
+            newIndex = PastePartsOfComposite(bindingsArray, bindings, index, actionName);
+            return newComposite;
+        }
+
+        public static SerializedProperty DuplicateElement(SerializedProperty arrayProperty, SerializedProperty toDuplicate, string name, int index, bool changeName = true)
+        {
+            var json = toDuplicate.CopyToJson(true);
+            return PasteElement(arrayProperty, json, index, out _, name, changeName);
+        }
+
+        #endregion
+
+        #region HelperMethods
+        private static List<SerializedProperty> GetBindingsForComposite(SerializedProperty bindingsArray, int indexOfComposite)
+        {
+            var compositeBindings = new List<SerializedProperty>();
+            var compositeStartIndex = InputActionSerializationHelpers.GetCompositeStartIndex(bindingsArray, indexOfComposite);
+            if (compositeStartIndex == -1)
+                return compositeBindings;
+
+            for (var i = compositeStartIndex + 1; i < bindingsArray.arraySize; ++i)
+            {
+                var bindingProperty = bindingsArray.GetArrayElementAtIndex(i);
+                var bindingFlags = (InputBinding.Flags)bindingProperty.FindPropertyRelative("m_Flags").intValue;
+                if ((bindingFlags & InputBinding.Flags.PartOfComposite) == 0)
+                    break;
+                compositeBindings.Add(bindingProperty);
+            }
+            return compositeBindings;
+        }
+
+        public static void EnsureUniqueName(SerializedProperty arrayElement)
```

Comment:
Just a thought/comment: Not sure if this a very "copy paste"-only helper method? It seems more generic even if only currently used by the `CopyPasterHelper` class. Same as for some other ones in this region (`AssignUniqueIDs`,etc).

Reply:
Reorganized some methods out of this region to the more generic InputActionSerializationHelpers class.